clinica/iotools/converters/adni_to_bids/adni_utils.py

```diff
@@ -650,17 +650,16 @@ def create_adni_sessions_dict(
             cprint(f"\tReading clinical data file: {location}")
 
             df_file = pd.read_csv(file_to_read_path, dtype=str)
-            df_filtered = filter_subj_bids(df_file, location, bids_ids)
+            df_filtered = filter_subj_bids(df_file, location, bids_ids).copy()
```

Comment:
Why is an explicit copy required?

Reply:
Could it be fixed in `filter_subj_bids` instead by returning `df.loc[condition, :]` instead of `df[conditon]`?

Reply:
I can look into it, but as the pattern to create a new column is the same as for the code below I suspect I'd need to explicitly copy at some point.

Reply:
I think it's best to have it fixed properly in what's returned by the function.

> the pattern to create a new column is the same as for the code below I suspect I'd need to explicitly copy at some point.

You could avoid it with [pandas.DataFrame.assign](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.assign.html).

Reply:
Good call. I fixed it in `filter_subj_bids` but it seems it doesn't solve the `SettingWithCopyWarning` because of later functions which would ideally require a different architecture (file reader for each clinical file as discussed).  I suggest to leave it in its current state for this PR and use a different PR to change architecture if needed.